packages/fx-core/src/component/generator/generator.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ContextV3 } from "@microsoft/teamsfx-api";
+import path from "path";
+import {
+  Component,
+  sendTelemetryEvent,
+  TelemetryEvent,
+  TelemetryProperty,
+} from "../../common/telemetry";
+import { SampleActionSeq, GenerateAction, TemplateActionSeq } from "./generateAction";
+import { GenerateContext } from "./generateContext";
+import {
+  genFileDataRenderReplaceFn,
+  genFileNameRenderReplaceFn,
+  getSampleInfoFromName,
+  getValidSampleDestination,
+  mergeReplaceMap,
+  sampleDefaultOnActionError,
+  templateDefaultOnActionError,
+} from "./utils";
+
+export class Generator {
+  public static async generateTemplate(
+    templateName: string,
+    language: string,
+    destinationPath: string,
+    ctx: ContextV3
+  ): Promise<void> {
+    const appName = ctx.projectSetting?.appName;
+    const projectId = ctx.projectSetting?.projectId;
+    const dataReplace = mergeReplaceMap(
+      {
+        appName: appName,
+        projectId: projectId,
+      },
+      ctx.templateReplace
+    );
+    const nameReplace = mergeReplaceMap(
+      {
+        appName: appName,
+      },
+      ctx.templateReplace
+    );
+    const destination = ctx.isAddFeature ? destinationPath : path.join(destinationPath, appName);
```

Comment:
I think there is already a stage property in Inputs. Usually, we check `inputs.stage === Stage.create`.
I prefer split this to two interfaces instead of control its behavior by an inner flag.

Reply:
Imagine which do you prefer:
```
createProject() {
  context.isAddFeature = false;
  Generator.generateTemplates(..., context);
}
addxxFeature() {
  context.isAddFeature = true;
  Generator.generateTemplates(..., context);
}
```
or 
```
createProject() {
  Generator.generateTemplates(...);
}
addxxFeature() {
  Generator.addBuildingBlocks(...);
}
```

Reply:
Will sync with you about this offline.